checkpatch_hook/__init__.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env python3
+# Copyright (c) Blu Wireless Technology Ltd. 2023, All Rights Reserved
+
+import argparse
+import importlib.resources
+import sys
+from collections import defaultdict
+from datetime import datetime
+from pathlib import Path
+from subprocess import PIPE, check_output, run
+from typing import DefaultDict
+
+import yaml
+
+# template used to add a patch-style header to a diff
+PATCH_TEMPLATE = """\
+From: A Non <a.non@bluwireless.com>
+Date: {current_datetime}
+Subject: [PATCH] {commit_msg}
+---
+{diff}
+--
+"""
+
+
+def _parse_args() -> dict:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--ci",
+        action="store_true",
+        help="Indicates that the (manual) hook is being run in CI",
+    )
+    parser.add_argument(
+        "files",
+        nargs="*",
+        type=Path,
+        default=[],
+        help="Files to check",
+    )
+    default_config_file = importlib.resources.files("checkpatch_hook.data").joinpath(
```

Comment:
I think I'd prefer this to be either indicated as a kind of constant, i.e. `DEFAULT_CONFIG_FILE` in the global context, or just loaded directly into the `default=` kwarg, instead of being a local variable in this function.